staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -1956,16 +1956,28 @@ type EnvVarResolverFunc func(e corev1.EnvVar) string
 // EnvValueFrom is exported for use by describers in other packages
 func EnvValueRetriever(pod *corev1.Pod) EnvVarResolverFunc {
 	return func(e corev1.EnvVar) string {
-		gv, err := schema.ParseGroupVersion(e.ValueFrom.FieldRef.APIVersion)
-		if err != nil {
-			return ""
-		}
-		gvk := gv.WithKind("Pod")
-		internalFieldPath, _, err := scheme.Scheme.ConvertFieldLabel(gvk, e.ValueFrom.FieldRef.FieldPath, "")
+		internalFieldPath, _, err := fieldpath.ConvertDownwardAPIFieldLabel(e.ValueFrom.FieldRef.APIVersion, e.ValueFrom.FieldRef.FieldPath, "")
 		if err != nil {
 			return "" // pod validation should catch this on create
 		}
 
+		switch internalFieldPath {
+		case "spec.nodeName":
+			return pod.Spec.NodeName
+		case "spec.serviceAccountName":
+			return pod.Spec.ServiceAccountName
+		case "status.hostIP":
+			return pod.Status.HostIP
+		case "status.podIP":
+			return pod.Status.PodIP
+		case "status.podIPs":
+			podIPs := make([]string, 0, len(pod.Status.PodIPs))
+			for _, podIP := range pod.Status.PodIPs {
+				podIPs = append(podIPs, podIP.IP)
+			}
+			return strings.Join(podIPs, ",")
+		}
+
 		valueFrom, err := fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
 		if err != nil {
 			return "" // pod validation should catch this on create
```

Comment:
it could be reasonable to render downward API env var values that can't be resolved in a better way than just an empty string